Bottom: c93e2238bbf040aee94664f94184a6137edded12
Top:    69f001f944f5ee8e89bd266c06934b7b7506cb74
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-05-17 17:43:35 +0800

Refresh of virtio-net-reset-num_buf-to-1

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 93ec469..910d2e7 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -536,6 +536,10 @@ static struct page *xdp_linearize_page(struct receive_queue *rq,
 		put_page(p);
 	}
 
+	/* receive_mergeable() assumes there's at least one buffer, so
+	 * align with it.
+	 */
+	*num_buf = 1;
 	/* Headroom does not contribute to packet length */
 	*len = page_off - VIRTIO_XDP_HEADROOM;
 	return page;
